src/modules/analytics-indexer/services/indexer.cron.service.ts

```diff
@@ -0,0 +1,215 @@
+import { generateLogMessage } from 'src/utils/generate-log-message';
+import { Lock } from '@multiversx/sdk-nestjs-common';
+import {
+    IndexerJob,
+    IndexerSession,
+    IndexerStatus,
+} from '../schemas/indexer.session.schema';
+import { PerformanceProfiler } from '@multiversx/sdk-nestjs-monitoring';
+import { CacheService } from '@multiversx/sdk-nestjs-cache';
+import { IndexerService } from './indexer.service';
+import { IndexerPersistenceService } from './indexer.persistence.service';
+import { IndexerEventTypes } from '../entities/indexer.event.types';
+
+const JOB_MAX_ATTEMPTS = 3;
+
+@Injectable()
+export class IndexerCronService {
+    constructor(
+        private readonly indexerService: IndexerService,
+        private readonly indexerPersistence: IndexerPersistenceService,
+        private readonly cachingService: CacheService,
+        @Inject(WINSTON_MODULE_PROVIDER) protected readonly logger: Logger,
+    ) {}
+
+    async onModuleInit() {
+        await this.resetStuckJobsAndSessions();
+    }
+
+    @Cron(CronExpression.EVERY_10_SECONDS)
+    @Lock({ name: 'indexAnalytics' })
+    public async indexAnalytics() {
+        let activeSession: IndexerSession;
+
+        try {
+            activeSession = await this.indexerPersistence.getActiveSession();
+            if (!activeSession) {
+                return;
+            }
+
+            const sessionAbortSignal = await this.getSessionAbortSignal(
+                activeSession,
+            );
+            if (sessionAbortSignal) {
+                return await this.markSessionAborted(activeSession);
+            }
+
+            if (activeSession.status === IndexerStatus.PENDING) {
+                activeSession.status = IndexerStatus.IN_PROGRESS;
+                await this.indexerPersistence.updateSession(activeSession);
+            }
+
+            await this.processIndexingJobs(activeSession);
+        } catch (error) {
+            const logMessage = generateLogMessage(
+                IndexerCronService.name,
+                this.indexAnalytics.name,
+                '',
+                error,
+            );
+            this.logger.error(logMessage);
+        }
+    }
+
+    private async processIndexingJobs(session: IndexerSession): Promise<void> {
+        for (const [index, job] of session.jobs.entries()) {
+            if (
+                job.status !== IndexerStatus.IN_PROGRESS &&
+                job.status !== IndexerStatus.PENDING
+            ) {
+                continue;
+            }
+
+            try {
+                job.status = IndexerStatus.IN_PROGRESS;
+
+                await this.indexerPersistence.updateSession(session);
+
+                session.jobs[index] = await this.runIndexingJob(
+                    job,
+                    session.eventTypes,
+                );
+
+                await this.indexerPersistence.updateSession(session);
+            } catch (error) {
+                await this.markSessionFailed(session);
```

Comment:
i would also save the error somewhere to be easier to debug. Or log it here

Reply:
done